src/modules/sdbus/watcher.c

```diff
@@ -0,0 +1,156 @@
+/************************************************************\
+ * Copyright 2023 Lawrence Livermore National Security, LLC
+ * (c.f. AUTHORS, NOTICE.LLNS, COPYING)
+ *
+ * This file is part of the Flux resource manager framework.
+ * For details, see https://github.com/flux-framework.
+ *
+ * SPDX-License-Identifier: LGPL-3.0
+\************************************************************/
+
+/* watcher.c - a flux watcher that becomes ready when sd-bus needs service
+ */
```

Comment:
as discussed offline, perhaps a small snippet to discuss whats going on.  Since many of the watchers don't take callbacks, I was confused what this was inititally doing.

Reply:
perhaps should also note, part of my initial confusion was that the `check_cb()` will be called even when a non-systemd related event (e.g. random other flux rpcs) trigger in libev land.  Should be fine, as `sd_bus_process()` should be fast, but that confused me as well.

Reply:
On further consideration, I think your intuition is correct: that the fd and timer watchers should call the bus watcher callback, not a check watcher.  That should eliminate the needless callbacks.  I'm not sure why I thought the check watcher would help.